src/main/scala/net/snowflake/spark/snowflake/SnowflakeConnectorUtils.scala

```diff
@@ -12,6 +12,22 @@ object SnowflakeConnectorUtils {
 
   @transient lazy val log = LoggerFactory.getLogger(getClass.getName)
 
+
+  /**
+    * Check Spark version, if Spark version matches SUPPORT_SPARK_VERSION enable PushDown, otherwise disable it.
+    */
+  private val SUPPORT_SPARK_VERSION = "2.2"
+
+  def checkSparkVersion(session: SparkSession): Boolean =
```

Comment:
Change this to "checkVersionAndEnablePushdown"

Reply:
done